packages/near-api-js/src/contract.ts

```diff
@@ -90,45 +119,92 @@ export class Contract {
     constructor(account: Account, contractId: string, options: ContractMethods) {
         this.account = account;
         this.contractId = contractId;
-        const { viewMethods = [], changeMethods = [] } = options;
-        viewMethods.forEach((methodName) => {
-            Object.defineProperty(this, methodName, {
+        const { viewMethods = [], changeMethods = [], abi: abiRoot } = options;
+
+        let viewMethodsWithAbi = viewMethods.map((name) => ({ name, abi: null as AbiFunction }));
+        let changeMethodsWithAbi = changeMethods.map((name) => ({ name, abi: null }));
+        if (abiRoot) {
+            const abiViewMethods = abiRoot.body.functions
+                .filter((m) => m.kind === AbiFunctionKind.View)
+                .map((m) => ({ name: m.name, abi: m }));
+            const abiChangeMethods = abiRoot.body.functions
+                .filter((methodAbi) => methodAbi.kind === AbiFunctionKind.Call)
+                .map((methodAbi) => ({ name: methodAbi.name, abi: methodAbi }));
+            viewMethodsWithAbi = viewMethodsWithAbi.concat(abiViewMethods);
+            changeMethodsWithAbi = changeMethodsWithAbi.concat(abiChangeMethods);
+        }
+
+        // Strict mode is disabled for now as it complains about unknown formats. We need to
+        // figure out if we want to support a fixed set of formats. `uint32` and `uint64`
+        // are added explicitly just to reduce the amount of warnings as these are very popular
+        // types.
+        const ajv = new Ajv({
+            strictSchema: false,
+            formats: {
+                uint32: true,
+                uint64: true
+            }
+        });
+        addFormats(ajv);
```

Comment:
What would you think about putting this in a private method that gets invoked here? If this is going to be extended over time it would be nice for it to have its own dedicated space, this constructor is already doing all the heavy lifting 😅